### ISSUE TITLE
Nullability of assigned field ignored

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#31394]
A null reference warning is reported for `a.F.ToString()`:
```c#
#nullable enable

class C<T> { }

class Program
{
    static void F(C<object>? x, C<object?> y)
    {
        var a = new { F = x };
        a = new { F = y }; // warning
        a.F.ToString(); // no warning expected
    }
}
```

**Comment by cston** [comment]
Same warning for `t.Item1.ToString()` below:
```c#
#nullable enable

class C<T> { }

class Program
{
    static void F(C<object>? x, C<object?> y)
    {
        var t = (x, y);
        t = (y, y); // warning
        t.Item1.ToString(); // no warning expected
    }
}
```
